### ISSUE TITLE
Overlapping tokens break visibility rules

### ISSUE THREAD

**Issue by night199uk** [lark-parser/lark#109]
Didn't get a chance to look into this myself yet. I have some instances where I need "." to be quiet, and some where I need it to be included, so I have this:
```
filename: file "." extension
DOT: "."
keepthis: blah DOT blah
```

Seems like the token compiler is deduping tokens. I either see file, Token(DOT, "."), extension, or all DOTs are hidden.

**Comment by erezsh** [comment]
Yes, this is a consequence of the current implementation. I can see why it might be counter-intuitive.
However, there are two workarounds:
1) Keep "." anonymous, and use `!keepthis: blah "." blah`
2) Keep "." anonymous, and use a rule: `keepthis: blah dot blah`

Having said that, it's probably better to separate identity and anonymity. Let's keep this issue open, so I can fix it next time I do a backward-incompatible release (probably along with fixing transformers).

**Comment by erezsh** [comment]
Fixed in 0.6.0
